pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -111,14 +217,58 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 	templateData.ExternalStaticDNS = config.BootstrapExternalStaticDNS
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
-	_, externalURLv6 := externalURLs(config.APIVIPs)
+	// If the user has manually set disableVirtualMediaTLS to False in the Provisioning CR, then enable TLS in ironic.
+	// The default value of 'true' is temporary (for backporting to older releases), which will be changed to 'false' in the future.
+	templateData.DisableIronicVirtualMediaTLS = true
+	protocol := "http"
+
+	provisioningCR := struct {
+		spec struct {
+			disableVirtualMediaTLS *bool `yaml:"disableVirtualMediaTLS"`
+		} `yaml:"spec"`
+	}{}
+	openshiftManifests := &manifests.Openshift{}
+	logrus.Debugf("Getting Openshift manifests dependency")
+	dependencies.Get(openshiftManifests)
+	logrus.Debugf("After getting Openshift manifests dependency")
+	var provisioningCRBytes []byte
+	for _, file := range openshiftManifests.Files() {
+		logrus.Debugf("openshiftManifests.Files() - file: %s", file.Filename)
+		if file.Filename == "openshift/99_baremetal-provisioning-config.yaml" {
+			logrus.Debugf("Found baremetal provisioning config file")
+			provisioningCRBytes = file.Data
+			break
+		}
+	}
+	if provisioningCRBytes != nil {
+		logrus.Debugf("Unmarshalling Provisioning CR")
+		yaml.Unmarshal(provisioningCRBytes, provisioningCR)
+		logrus.Debugf("provisioningCR: %v", provisioningCR)
+	}
+	if provisioningCR.spec.disableVirtualMediaTLS != nil {
+		logrus.Debugf("provisioningCR.spec.disableVirtualMediaTLS != nil")
+		templateData.DisableIronicVirtualMediaTLS = *provisioningCR.spec.disableVirtualMediaTLS
+		if !*provisioningCR.spec.disableVirtualMediaTLS {
+			logrus.Debugf("provisioningCR.spec.disableVirtualMediaTLS is false")
+			protocol = "https"
+			installConfig := &installconfig.InstallConfig{}
+			logrus.Debugf("Getting installConfig dependency")
+			dependencies.Get(installConfig)
+			err := generateIronicVMediaTLSCertificate(installConfig.Config.Platform.BareMetal.BootstrapProvisioningIP)
+			if err != nil {
+				logrus.Debugf("Error in creating TLS certificate for ironic virtual media: ", err) // I should probably do something more than this
```

Comment:
At the very least, it should not be a debug message

Reply:
Updated it to an Error